openedx/adg/lms/applications/views.py

```diff
@@ -135,3 +138,96 @@ def get_context_data(self, **kwargs):
+
+    def is_precondition_satisfied(self):
+        """
+        Checks if a written application is already submitted or not.
+
+        Returns:
+            Boolean, True or False.
+        """
+        user_application_hub, _ = ApplicationHub.objects.get_or_create(user=self.request.user)
+
+        return not user_application_hub.is_written_application_completed
+
+    def handle_no_permission(self):
+        """
+        Redirects to application hub on get request or returns htt 400 on post request.
+
+        Returns:
+            HttpResponse object.
+        """
+        if self.request.method == 'POST':
+            return HttpResponse(status=400)
+        else:
+            return redirect('application_hub')
+
+    def get(self, request):
+        """
+        Send the context data for example a list of business lines and saved user_application.
+
+        Returns:
+            HttpResponse object.
+        """
+        business_lines = BusinessLine.objects.all()
+        file_name = None
+
+        try:
+            user_application = UserApplication.objects.get(user=request.user)
+            file_name = Path(user_application.cover_letter_file.name).name
+        except UserApplication.DoesNotExist:
+            user_application = None
+
+        context = {
+            'business_lines': business_lines,
+            'logo_url': 'http://localhost:18000/media/',
+            'user_application': user_application,
+            'csrf_token': get_token(request),
+            'filename': file_name
+        }
+
+        return render(request, self.template_name, context)
+
+    def post(self, request):
+        """
+        Submit user application.
+
+        Returns:
+            HttpResponse object.
+        """
+        user_application, _ = UserApplication.objects.get_or_create(user=request.user)
+
+        if 'business_line' in request.POST and request.POST['business_line']:
+            business_line = BusinessLine.objects.get(id=request.POST['business_line'])
+            user_application.business_line = business_line
+            user_application.save()
+
+        if 'text-coverletter' in request.POST:
+            if request.POST['text-coverletter']:
+                cover_letter_text = request.POST['text-coverletter']
+                user_application.cover_letter = cover_letter_text
+            else:
+                user_application.cover_letter = None
+
+            user_application.cover_letter_file = None
+            user_application.save()
+        elif 'add-coverletter' in request.FILES:
+            cover_letter_file = request.FILES['add-coverletter']
+            user_application.cover_letter_file = cover_letter_file
+            user_application.cover_letter = None
+            user_application.save()
+
+        if request.POST['next'] == 'back':
+            return redirect('application_experience')
+        else:
+            application_hub, _ = ApplicationHub.objects.get_or_create(user=request.user)
```

Comment:
Do we need `get_or_create` here? Or will `get` suffice?
Also, we can access the ApplicationHub with related name on the user object as `user.application_hub`